tests/functional/tdtagged/td1dtagged.py

```diff
@@ -132,12 +240,114 @@ def vthz(x):
             )
 
 
+def make_figure():
+    from scipy.optimize import curve_fit
+    from pyphare.pharesee.hierarchy import finest_field
+
+    rwT    = Run("./withTagging")
+    rNoRef = Run("./noRefinement")
+
+    plot_time = 11
+    v = 2
+
+    BwT = rwT.GetB(plot_time)
+    BNoRef = rNoRef.GetB(plot_time)
+    JwT = rwT.GetJ(plot_time)
+    JNoRef = rNoRef.GetJ(plot_time)
+
+    bywT, xbywT  = finest_field(BwT, "By")
+    byNoRef, xbyNoRef  = finest_field(BNoRef, "By")
+    jzwT, xjzwT  = finest_field(JwT, "Jz")
+    jzNoRef, xjzNoRef  = finest_field(JNoRef, "Jz")
+
+    fig, axarr = plt.subplots(nrows=3, figsize=(8,8))
+
+    def S(x, x0, l):
+        return 0.5*(1+np.tanh((x-x0)/l))
+
+    def by(x):
+        L=200
+        v1=-1
+        v2=1
+        return v1 + (v2-v1)*(S(x, L*0.25, 1)-S(x, L*0.75, 1))
+
+    ax0, ax1, ax2 = axarr
+    ax0.plot(xbywT, bywT)
+    ax0.plot(xbyNoRef, byNoRef, color="k", alpha=0.6)
+    ax0.plot(xbyNoRef, by(xbyNoRef), ls='--')
+
+    ax1.plot(xbywT, bywT)
+    ax1.plot(xbyNoRef, byNoRef, color='k')
+    ax1.set_xlim((wT0,195))
+    ax1.set_ylim((-1.5, 2))
+
+    ax2.plot(xjzwT, jzwT)
+    ax2.plot(xjzNoRef, jzNoRef, color='k')
+    ax2.set_xlim((wT0,195))
+    ax2.set_ylim((-1.5, 0.5))
+
+
+    # draw level patches
+    for ilvl,level in BwT.levels().items():
+        for patch in level.patches:
+            dx = patch.layout.dl[0]
+            x0 = patch.origin[0]
+            x1 = (patch.box.upper[0]+1)*patch.layout.dl[0]
+            for ax in (ax1, ax2, ax0):
+                ax.axvspan(x0, x1, color='b',ec='k', alpha=0.2,
+                            ymin=ilvl/4, ymax=(ilvl+1)/4)
+
+
+    from pyphare.pharesee.plotting import zoom_effect
+    zoom_effect(ax0, ax1, wT0, 195)
+
+    for ax in (ax0, ax1, ax2):
+        ax.axvline(wT0+plot_time*v, color="r")
+
+
+    # select data around the rightward TD
+    idx = np.where((xby15>150) & (xby15<190))
+    xx = xby15[idx]
+    bby = by15[idx]
+
+
+    # now we will fit by_fit to the data
+    # and we expect to find x0=172 and L=1
+    # or close enough
+    def by_fit(x, x0,L):
+        v1=1
+        v2=-1
+        return v1 + (v2-v1)*S(x, x0, L)
+
+    popt,pcov = curve_fit(by_fit, xx, bby, p0=(150,1))
+    x0, L = popt
+
+    if np.abs(L-1)>0.5:
+        raise RuntimeError(f"L (={L}) too far from 1.O")
+    if np.abs(x0-(150+plot_time*v))>0.5:
+        raise RuntimeError(f"x0 (={x0}) too far from 172")
+
+
+
+
+
 
 def main():
-    config()
+    from pybindlibs import mpi_rank
```

Comment:
this works?

Reply:
why not?

Reply:
![image](https://user-images.githubusercontent.com/30491/114551017-22888600-9c63-11eb-8e65-4f1222ffea8f.png)

Reply:
ok
```python
from pybindlibs.cpp import mpi_rank
```
should do it